### PR TITLE
Add 1.X server compatibility with allow_unencrypted attribute

### LINKED ISSUE #83: disables encryption and allows use of 1.x server?
### Cookbook version
2.6.4
push-jobs-client.x86_64        2.1.0-1.el7                         installed

### Chef-client version
12.9.41

### Platform Details
Oracle Linux Server release 7.1

### Scenario:
failing to connect to 1.x (?) Chef Server

### Steps to Reproduce:

The following error started to show up on a node (I tried ```push-jobs``` versions from ```2.6.4``` to ```2.8.0``` ...)

I tried to override:

```
default['push_jobs']['allow_unencrypted'] = true
```
but reading the source

https://github.com/chef-cookbooks/push-jobs/blob/master/templates/default/push-jobs-client.rb.erb

the ```allow_unencypted``` value is not processed.

Is there a component I'm to roll back to overcome this?

Thanks

### Expected Result:
opscode-push-jobs service to connect to Chef Server

### Actual Result:
Seeing following log entries (repeated every few seconds):
```
2016-05-23_15:55:29.96948 INFO: [server1] using config file path: '/etc/chef/push-jobs-client.rb'
2016-05-23_15:55:29.97089 INFO: [server1] Using node name: server1
2016-05-23_15:55:29.97092 INFO: [server1] Using org name: myorg
2016-05-23_15:55:29.97093 INFO: [server1] Using Chef server: https://chefserver:443/organizations/myorg
2016-05-23_15:55:29.97096 INFO: [server1] Using private key: /etc/chef/client.pem
2016-05-23_15:55:29.97097 INFO: [server1] Incarnation ID: 162c6f0d-8e4f-395f-b107-ca8b860036fe
2016-05-23_15:55:29.97097 INFO: [server1] Allowing fallback to unencrypted connection: false
2016-05-23_15:55:29.97098 INFO: [server1] Starting client ...
2016-05-23_15:55:29.97099 INFO: [server1] Retrieving configuration from https://chefserver:443/organizations/myorg//pushy/config/server1: ...
2016-05-23_15:55:30.03089 ERROR: [server1] Exiting: No key returned from server; server may be using 1.x protocol. The config flag 'allow_unencrypted' disables encryption and allows use of 1.x server. Use with caution!
2016-05-23_15:55:30.03092 [server1] Exiting: No key returned from server; server may be using 1.x protocol. The config flag 'allow_unencrypted' disables encryption and allows use of 1.x server. Use with caution!
```

### PR DESCRIPTION
**Pull request by tas50**
### Description

Add the ability to set the allow_unencrypted attribute in the push-jobs config.

### Issues Resolved

#83 

### Check List
- [ ] All tests pass. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] The CLA has been signed. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD

### CODE REVIEW

File: spec/unit/recipes/install_spec.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
   context 'without package_url' do
     let(:chef_run) do
-      runner = ChefSpec::SoloRunner.new(platform: 'ubuntu', version: '10.04')
+      runner = ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '10.04')
```
Comment (mmzyk) [inline spec/unit/recipes/install_spec.rb]: Did you intent this to be 14.04 as well?

**Comment by mmzyk** [comment]
Looks good to me, assuming tests pass. One comment on one of the updates to 14.04 where it looks like it might have been left at 10.04 by accident.

**Comment by tas50** [comment]
Fixed the last 10.04

**Comment by gliptak** [comment]
Thank you, this corrected the 1.x error.

**Pull request merged**